homeassistant/components/screenlogic/__init__.py

```diff
@@ -88,6 +91,95 @@ async def async_update_listener(hass: HomeAssistant, entry: ConfigEntry) -> None
     await hass.config_entries.async_reload(entry.entry_id)
 
 
+async def _async_migrate_entries(
+    hass: HomeAssistant, config_entry: ConfigEntry
+) -> None:
+    """Migrate to new entity names."""
+    entity_registry = er.async_get(hass)
+
+    for entry in er.async_entries_for_config_entry(
+        entity_registry, config_entry.entry_id
+    ):
+        try:
+            source_mac, source_key = entry.unique_id.split("_", 1)
+        except ValueError:
+            _LOGGER.debug("Unable to parse unique_id '%s'", entry.unique_id)
+            continue
+
+        source_index = None
+        if (
+            len(key_parts := source_key.rsplit("_", 1)) == 2
+            and key_parts[1].isdecimal()
+        ):
+            source_key, source_index = key_parts
+
+        _LOGGER.debug(
+            "Checking migration status for '%s' against key '%s'",
+            entry.unique_id,
+            source_key,
+        )
```

Comment:
Usually we write end to end tests for entity unique id migrations

Reply:
How extensive should this be?

Should I just be testing `_async_migrate_entries()` or the entire `async_setup_entry()` all the way through `async_forward_entry_setups()`?  Asking because I'm wondering if I need to create a full fake-connected ScreenLogicGateway instance.

Also, is there a coverage % requirement?

Reply:
The coverage needs to be for the migrate code, but we need to write the test to HA standards so that we don't interact with the integration internals. To accomplish that we should use the standard interfaces to setup the config entry `await hass.config_entries.async_setup(config_entry.entry_id)` and make sure not to access `hass.data` in the test.

https://developers.home-assistant.io/docs/development_testing?_highlight=testing#writing-tests-for-integrations

It can be as simple as something like #95656 (there will be a bit more mocking of the screen logic gateway here)

Reply:
I think I've hit everything required.